crates/services/producer/src/block_producer/tests.rs

```diff
@@ -203,12 +210,44 @@ async fn production_fails_on_execution_error() {
     );
 }
 
+// TODO: Add test that checks the gas price on the mint tx after `Executor` refactor
+//   https://github.com/FuelLabs/fuel-core/issues/1751
+#[ignore]
+#[tokio::test]
+async fn produce_and_execute_block_txpool__includes_gas_price_provided_on_mint_tx() {
+    todo!()
+    // // given
+    // let gas_price = 1_000;
+    // let gas_price_provider = StaticGasPrice::new(gas_price);
+    // let ctx = TestContext::default();
+    // let producer = ctx.producer_with_gas_price_provider(gas_price_provider);
+    //
+    // // when
+    // let changes = producer
+    //     .produce_and_execute_block_txpool(1u32.into(), Tai64::now(), 1_000_000_000)
+    //     .await
+    //     .unwrap();
+    //
+    // // then
+    // let mint_tx = changes.get_mint_tx();
+    // assert_eq!(mint_tx.gas_price, gas_price);
+}
+
+// TODO: Add test that checks the failure case for gas_price after `Executor` refactor
```

Comment:
I didn't get why we can't implement this test now. Could you elaborate more on it, please?=)

Reply:
You're right. I've implemented it.